### PR TITLE
Adding support for zbarlight

### PR DESCRIPTION
**Pull request by GoBig87**
Added libzbar and zbarlight recipe and also fixed the kivy recipe to turn on the camera AVfoundation for ios and some other tweeks to get the camera working.

### CODE REVIEW

File: recipes/kivy/__init__.py [hunk 1]
```diff
@@ -1,16 +1,23 @@
-from toolchain import CythonRecipe
+from toolchain import CythonRecipe, shprint
```
Comment (AndreMiras) [inline recipes/kivy/__init__.py]: No sure we need that `shprint` import

File: recipes/zbarlight/__init__.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+                _env=build_env)
+        self.apply_patch("zbarlight_1_2.patch")#Issue getting the version, hard coding for now
+        self.biglink()
+
+    def install(self):
+        arch = list(self.filtered_archs)[0]
+        build_dir = join(self.get_build_dir(arch.arch),'build','lib.macosx-10.13-x86_64-2.7','zbarlight')
+        dist_dir  = join(self.ctx.dist_dir,'root','python','lib','python2.7','site-packages','zbarlight')
+        #Patch before Copying
+        #self.apply_patch("zbarlight_1_2.patch")#Issue getting the version, hard coding for now
```
Comment (AndreMiras) [inline recipes/zbarlight/__init__.py]: Minor: why is this patch line commented rather than being removed? The patch is already applied in `build_arch()`

**Comment by AndreMiras** [comment]
It's looking good overall, thanks :heart:  I've added a couple of minor comments, but I think it could be merged as it is.

File: recipes/zbarlight/__init__.py (hunk 2 above)
Comment (GoBig87) [reply recipes/zbarlight/__init__.py]: Oh I was trying to patch it after the file was moved.  It was easier to just patch the file before it was moved and I must have forgot to remove the line once I got the patch working before the move

Comment (AndreMiras) [reply recipes/zbarlight/__init__.py]: yep makes sense. You don't need to close the pull request if it's a little change. Simply update the pull request.
```sh
vim recipes/zbarlight/__init__.py
git add recipes/zbarlight/__init__.py
git commit --amend
git push
```

**Comment by akshayaurora** [comment]
Please open independent prs for each recipes. Separate out the avfoundation + kivy camera

**Comment by akshayaurora** [comment]
@GoBig87 could you please separate out the AVFoundation & Camera Fixes into a separate PR

**Comment by GoBig87** [comment]
Sorry,  I've been busy spending every moment of free time trying to figure out how to add obj-c enabled swift frameworks into kivy-ios project.  I plan to clean up my pull requests once I get this figured out.

**Comment by GoBig87** [comment]
Sorry I haven't forgotten about this issue.  I've finally finished my app and its ready for beta for ios and android.  I want to try and fix an issue with the camera presenting the wrong colors before I submit the camera changes pull request.  For some reason the Red and Blue color is switched out and I'm going to fix it before I submit the camera pull request.

**Comment by AndreMiras** [comment]
Good the hear, I think it's still a good idea to pull request the zbar part as a separated pull request.

**Comment by GoBig87** [comment]
Adding the pull request (https://github.com/kivy/kivy/pull/6156) to fix the issue in Kivy-master rather than making patches inside kivy-ios.

**Comment by GoBig87** [comment]
Moving and separating pull requests to https://github.com/kivy/kivy-ios/pull/359 and https://github.com/kivy/kivy/pull/6156.

**Pull request closed without merging**